src/main/java/mate/jdbc/dao/impl/FlightDaoImpl.java

```diff
@@ -0,0 +1,130 @@
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection
+                         .prepareStatement(query, Statement.RETURN_GENERATED_KEYS)
+        ) {
+            statement.setString(1, flight.getNumber());
+            statement.setString(2, flight.getCarrier());
+            statement.setLong(3, flight.getManufacturer());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                flight.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert flight to DB : "
+                    + flight, e);
+        }
+        return flight;
+    }
+
+    @Override
+    public Optional<Flight> get(Long id) {
+        String query = "SELECT * FROM flights WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)
+        ) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getFlight(resultSet));
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get flight from DB by id = " + id, e);
+        }
+    }
+
+    @Override
+    public List<Flight> getAll() {
+        List<Flight> flights = new ArrayList<>();
+        String query = "SELECT * FROM flights WHERE is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 Statement statement = connection.createStatement()
+        ) {
+            ResultSet resultSet = statement.executeQuery(query);
+            while (resultSet.next()) {
+                flights.add(getFlight(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all flights from DB", e);
+        }
+        return flights;
+    }
+
+    @Override
+    public Flight update(Flight flight) {
+        String query = "UPDATE flights SET  number = ?, carrier = ?, manufacturer_id = ?"
+                + " WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)
+        ) {
+            statement.setString(1, flight.getNumber());
+            statement.setString(2, flight.getCarrier());
+            statement.setLong(3, flight.getManufacturer());
+            statement.setObject(4, flight.getId());
+            if (statement.executeUpdate() > 0) {
+                return flight;
+            } else {
```

Comment:
we don't need else after return

Reply:
for example, when we will id, then bigger that we have in table

Reply:
I understand, but we don't need else block after we have return keyword or we throw exception in if. If we go inside of If code will stop execution after return or throwing exception.